dizoo/atari/example/atari_dqn_dist_rdma.py

```diff
@@ -26,46 +29,73 @@ def main():
             task.use(
-                context_exchanger(
-                    send_keys=["train_iter"],
-                    recv_keys=["trajectories", "episodes", "env_step", "env_episode"],
+                ContextExchanger(
+                    # send_keys=["train_iter"],
+                    # recv_keys=["trajectories", "episodes", "env_step", "env_episode"],
                     skip_n_iter=0
                 )
             )
-            task.use(model_exchanger(model, is_learner=True))
+            task.use(ModelExchanger(model))
             task.use(nstep_reward_enhancer(cfg))
             task.use(data_pusher(cfg, buffer_))
             task.use(OffPolicyLearner(cfg, policy.learn_mode, buffer_))
             task.use(CkptSaver(cfg, policy, train_freq=1000))
+            # Because we are timing on the learner side, in order to prevent the learner from starting
+            # too fast and waiting for the collector initialization to complete during timing,
+            # we let the learner sleep for a period of time (fix me).
+            # sleep(40)
 
         elif 'collector' in task.router.labels:
+            policy = DQNPolicy(cfg.policy, model=model)
             logging.info("Collector running on node {}".format(task.router.node_id))
             collector_cfg = deepcopy(cfg.env)
             collector_cfg.is_train = True
+            logging.info(cfg.env.manager)
+            logging.info(type(cfg.env.manager))
+            #task.router.judge_use_cuda_shm(cfg)
+            logging.debug("cuda_shared_memory {}".format(cfg.env.manager.cuda_shared_memory))
             collector_env = SubprocessEnvManagerV2(
                 env_fn=[lambda: AtariEnv(collector_cfg) for _ in range(cfg.env.collector_env_num)], cfg=cfg.env.manager
             )
             task.use(
-                context_exchanger(
-                    send_keys=["trajectories", "episodes", "env_step", "env_episode"],
-                    recv_keys=["train_iter"],
+                ContextExchanger(
+                    # send_keys=["trajectories", "episodes", "env_step", "env_episode"],
+                    # recv_keys=["train_iter"],
                     skip_n_iter=1
                 )
             )
-            task.use(model_exchanger(model, is_learner=False))
+            task.use(ModelExchanger(model))
             task.use(eps_greedy_handler(cfg))
             task.use(StepCollector(cfg, policy.collect_mode, collector_env))
             task.use(termination_checker(max_env_step=int(1e7)))
         else:
             raise KeyError("invalid router labels: {}".format(task.router.labels))
 
-        task.run()
+        start_time = task.run(max_step=100)
+        end_time = time.time()
+        logging.debug("atari iter 99 use {:.4f} s,".format(end_time - start_time))
+
+        if 'collector' in task.router.labels:
+            time.sleep(20)  # give some time to learner recving
```

Comment:
can we implement this wait mechanism more elegantly?

Reply:
Done, add `wait_for_shutdown()`  and ‘stop()’ to exit gracefully.